### PR TITLE
#307: Single Item extension - Providers and License on the item level

### LINKED ISSUE #307: Item-level license and providers
For `Item`s that are difficult to group into coherent `Collection`s, we should provide a clearly-defined mechanism for specifying their `license` and `provider`s.

In other words, `license` and `providers` are important enough properties that they should be calculable for a given item, either directly or via the `Collection` that it belongs to. `Item`s that only belong to `Catalog`s are currently orphaned with no license data.

OpenAerialMap is a collection of heterogeneously-licensed data (differing CC licenses) provided by a variety of users. In order to surface licensing data for individual items, a `Catalog` would need to consist of multiple `Collection`s per user, one for each license used by that user. To further complicate things, legacy imagery does not include user associations. Complicating things even *more*, user uploads may have varying [free-form] attribution.

From Gitter:

@cholmes:

>  for OAM I'd have many collections. It can be one 'catalog' at the highest level, and then you'd navigate down to like the user level, and perhaps each user (or user + specific drone?) would be a 'collection'.

@matthewhanson:

> The license thing had come up before, when Jeff at DG mentioned license might vary in a collection.
> I think for OAM there's lots of other good reasons to make different collections for the data (varying platforms, instruments, spectral band definitions, etc), however for some collections it is useful to group together things from multiple sources. I'm currently working on making a collection that represents a specific area, with mixed imagery as member Items, so the license thing is pertinent there as well (it could be DG imagery, or publicly available imagery).
> We could solve this pretty easily by moving license into the Item properties. It could then be specified using the Commons extension in the Collection, or not and just specified per Item.

@m-mohr:

> You are taking a subset of a collection and merge it to another collection, so to me it would be clear that there should be multiple collections involved.
> If we are stripping away license from collections for that reason, we would also need to strip provider. Your combined imagery probably doesn't only have different licenses, but also different providers. And basically, the same applies for the extents. So we are basically back to discussing usefulness of collections as we would strip away all fields that separate collections from catalogs. Really? We just discussed that in depth...

### PR DESCRIPTION
**Pull request by m-mohr**
That extension name ([mixin](https://en.wikipedia.org/wiki/Mixin), prefix: `mix`) is a bit... strange(?), but describes basically what we are doing here:
* using fields from collections as mixin and
* mixing providers and licenses in a collection ;-)

Additionally, it is pretty generic and allows to extend it with new fields in future if we like.

Fixes #307.

Further changes in #303 would most likely need to be adopted here.

### CODE REVIEW

File: extensions/mixin/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Provides a way to specify several fields in Items that usually reside on the collection-level such as license and provider.
+
+- [Example](example.json)
+- [JSON Schema](schema.json)
+
+## Item fields
+
+| Field Name       | Type              | Description                                                  |
+| ---------------- | ----------------- | ------------------------------------------------------------ |
+| mix:license      | string            | Items's license(s) as a SPDX [License identifier](https://spdx.org/licenses/) or [expression](https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60) or `proprietary` if the license is not on the SPDX license list. Proprietary licensed data SHOULD add a link to the license text, see the `license` relation type. |
```
Comment (mojodna) [inline extensions/mixin/README.md]:
```suggestion
| mix:license      | string            | Item's license(s) as a SPDX [License identifier](https://spdx.org/licenses/) or [expression](https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60) or `proprietary` if the license is not on the SPDX license list. Proprietary licensed data SHOULD add a link to the license text, see the `license` relation type. |
```

File: extensions/mixin/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+The object provides information about a provider. A provider is any of the organizations that captured or processed the content of the collection and therefore influenced the data offered by this collection. May also include information about the final storage provider hosting the data.
+
+| Field Name  | Type      | Description                                                  |
+| ----------- | --------- | ------------------------------------------------------------ |
+| name        | string    | **REQUIRED.** The name of the organization or the individual. |
+| description | string    | Multi-line description to add further provider information such as processing details for processors and producers, hosting details for hosts or basic contact information. [CommonMark 0.28](http://commonmark.org/) syntax MAY be used for rich text representation. |
+| roles       | [string]  | Roles of the provider. Any of `licensor`, `producer`, `processor` or `host`. |
+| url         | string    | Homepage on which the provider describes the dataset and publishes contact information. |
+
+**type**: The type of the provider can be one of the following elements:
```
Comment (mojodna) [inline extensions/mixin/README.md]:
```suggestion
**roles**: The provider's role(s) can be one or more of the following elements:
```

File: extensions/mixin/example.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      ]
+    ]
+  },
+  "properties":{
+    "datetime":"2018-01-01T13:21:30Z",
+    "mix:license":"PDDL-1.0",
+    "mix:providers":[
+      {
+        "name": "John Doe",
+        "type": "producer",
```
Comment (mojodna) [inline extensions/mixin/example.json]:
```suggestion
        "roles": ["producer"],
```

File: extensions/mixin/README.md (hunk 2 above)
Comment (m-mohr) [reply extensions/mixin/README.md]: Fixed, in both extension and collection spec.

File: extensions/mixin/example.json (hunk 3 above)
Comment (m-mohr) [reply extensions/mixin/example.json]: Fixed.

File: extensions/mixin/README.md (hunk 1 above)
Comment (m-mohr) [reply extensions/mixin/README.md]: Fixed, thanks.

**Comment by m-mohr** [comment]
Fixed the type => roles issues - were copy&paste errors as #303 was not merged yet.

I'm always trying to make this extensible. We are not introducing the mixin concept itself, but leave the door open to add it later. For now we limit it to license and provider. That said, I personally would prefer to deprecate this extension anyway in 0.7.0 based on the planned improvements to the metadata model.

We are recommending prefixes in our spec, so I'd say we should follow our own recommendation. Nevertheless, if you feel it should be without prefixes, I think I'm fine with it.

**Comment by m-mohr** [comment]
Maybe we could also go with the prefix "item" and "Single Item Extension"?! That would also make a clearer distinction between the collection provider & license and the item provider and license. Maybe that's a better solution than mixin...

**Pull request merged**